echidna/core/smear.py

```diff
@@ -389,3 +389,32 @@ def weight_gaussian_radius_spectra(self, true_spectrum, num_sigma=5.):
                         i += 1
         smeared_spectrum._raw_events = raw_events
         return smeared_spectrum
+
+
+class EResSmear(Smear):
+    """ Allows you to smear directly by energy resolution (e.g. 0.05 * E)
+
+    Inherits from :class:`Smear` to provide exactly the same functionality
+    but overrides :meth:`Smear.get_energy_sigma` to use an energy 
+    resolution to calculate sigma.
+
+    Attributes:
+      _light_yield (float): Number of PMT hits expected for a
+        MeV energy deposit in NHit/MeV
+      _energy_resolution (float): energy resolution e.g. 0.05 for 5%. 
+      _position_resolution (float): Sigma in mm
+    """
+    def __init__(self, energy_resolution):
+        super(EResSmear, self).__init__()
+        self._energy_resolution = energy_resolution
+
+    def get_energy_sigma(self, energy):
+        """ Calculates sigma at a given energy.
+
+        Args:
+          energy (float): Energy value of data point(s)
+
+        Returns:
+          float. Sigma equivalent to energy * energy_resolution
+        """
+        return energy * self._energy_resolution
```

Comment:
The EResSmear class is the only thing I dont understand in this PR. The energy resolution is a function of energy itself. There is a function in the smear class already which calculates the sigma line 82:
```python
def get_energy_sigma(self, energy):
         """ Calculates sigma at a given energy.
 
         Args:
           energy (float): Energy value of data point(s)
 
         Returns:
           Sigma equivalent to sqrt(energy/_light_yield)
         """
         return np.sqrt(energy/self._light_yield)
```
I'm struggling to think of a case where this new sigma is applicable?
Also, if this extra form of sigma is required, does it require a new class?